src/Cake.Common/Tools/NuGet/Update/NuGetUpdateSettings.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System.Collections.Generic;
+using Cake.Core.IO;
+
+namespace Cake.Common.Tools.NuGet.Update
+{
+    /// <summary>
+    /// Contains settings used by <see cref="NuGetUpdater"/>.
+    /// </summary>
+    public sealed class NuGetUpdateSettings
+    {
+        /// <summary>
+        /// Gets or sets the path to <c>nuget.exe</c>.
+        /// </summary>
+        public FilePath ToolPath { get; set; }
+
+        /// <summary>
+        /// Gets or sets the package ids to update.
+        /// </summary>
+        /// <value>The package ids to update.</value>
+        public ICollection<string> Id { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether to look for updates with the highest 
+        /// version available within the same major and minor version as the installed package.
+        /// </summary>
+        /// <value>
+        ///   <c>true</c> if safe; otherwise, <c>false</c>.
+        /// </value>
+        public bool Safe { get; set; }
+
```

Comment:
Add optional `ICollection<string>` `Source` property to be able to specify alternative sources for update.
Like the `NugetRestoreSettings` i.e.
https://github.com/cake-build/cake/blob/develop/src/Cake.Common/Tools/NuGet/Restore/NugetRestoreSettings.cs#L32